mne_bids/write.py

```diff
@@ -698,16 +698,30 @@ def _write_raw_brainvision(raw, bids_fname, events):
     meas_date = raw.info['meas_date']
     if meas_date is not None:
         meas_date = _stamp_to_dt(meas_date)
-    # writing with float32 and resolution 1e-9 should be precise enough
-    # to avoid loss of resolution for any kind of data.
+
+    if raw.orig_format == 'single':
+        fmt = 'binary_float32'
+    elif raw.orig_format == 'short':
+        fmt = 'binary_int16'
+    elif raw.orig_format == 'int':
+        raise RuntimeError('pybv currently does not support writing INT16 '
```

Comment:
```suggestion
        raise RuntimeError('pybv currently does not support writing INT32 '
```

int16 would be short as above